app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -77,7 +77,12 @@ public void onClick(final View view) {
         }
 
         final ListView lv = (ListView) findViewById(R.id.listView);
-        lv.setAdapter(new ContactListAdapter(this, this.contacts));
+        try {
+            fillDatabaseWithDemoData();
+            lv.setAdapter(new ContactListAdapter(this, serviceLocator.getDatabase().getAllContacts()));
+        } catch (final IOException e) {
+            LOGGER.error("Failed to retrieve contacts from database", e);
```

Comment:
I agree, but the `fillDatabaseWithDemoData();` should be removed in the future, thats why I exluded it from the log